### DISCUSSION TITLE
[discussion] Convenient lambda function calls

### DISCUSSION THREAD

**Discussion by fawdlstty** [dotnet/csharplang#3845] [General]
Direct calls to lambda functions weren't obvious before, but with the new Switch syntax, the need has become urgent.It is often necessary to execute a block of code after hitting a condition.The existing implementation method is as follows:

```csharp
return n switch {
    1 => true,
    2 => false,
    _ => ((Func<bool>)(() => {
        // do something...
        return true;
    })).Invoke (),
};
```

propose:

```csharp
return n switch {
    1 => true,
    2 => false,
    _ => () => {
        // do something...
        return true;
    } (),
};
```

**Comment by YairHalberstadt** [comment]
Expression blocks would be a much more performant and simple way to achieve the same thing.

**Comment by fawdlstty** [comment]
@YairHalberstadt More code is needed to do the same thing

**Comment by YairHalberstadt** [comment]
@fawdlstty 

What do you mean? With expression blocks this would be:

```csharp
return n switch {
    1 => true,
    2 => false,
    _ => {
        // do something...
        true
    },
};
```

**Comment by YairHalberstadt** [comment]
See https://github.com/dotnet/csharplang/issues/3086

**Comment by fawdlstty** [comment]
You are right, but I need more people's opinions

**Comment by theunrepentantgeek** [comment]
Why do we need more syntax here?

Both local functions and the original **switch** statement seem like they'd be natural ways to express the need.

Shoehorning a **switch** expression in when you don't have expressions for the branches seems like an odd choice to me.

(Both of these would also avoid the allocations required by use of lambdas.)
